src/com/phonebookservice/util/MyArrayList.java

```diff
@@ -392,4 +428,26 @@ private boolean checkIfEqual(final Object element, final int index) {
                 || this.data[index] != null && data[index].equals(element);
 
     }
+
+    private void checkCollectionNotNull(final Collection<?> collection) {
+        if (collection == null) {
+            throw new NullPointerException("collection is null");
+        }
+    }
+
```

Comment:
What is the complexity of this method?

Could you do it better ?

Reply:
complexity: nm^2
yes

Reply:
Can you do better ? Please discuss here?